sky/utils/controller_utils.py

```diff
@@ -187,79 +187,54 @@ def from_type(cls, controller_type: str) -> Optional['Controllers']:
 
 # Install cli dependencies. Not using SkyPilot wheels because the wheel
 # can be cleaned up by another process.
-# TODO(zhwu): Keep the dependencies align with the ones in setup.py
 def _get_cloud_dependencies_installation_commands(
         controller: Controllers) -> List[str]:
-    # TODO(tian): Make dependency installation command a method of cloud
-    # class and get all installation command for enabled clouds.
-    commands = []
     # We use <step>/<total> instead of strong formatting, as we need to update
     # the <total> at the end of the for loop, and python does not support
     # partial string formatting.
     prefix_str = ('[<step>/<total>] Check & install cloud dependencies '
                   'on controller: ')
+    commands: List[str] = []
     # This is to make sure the shorter checking message does not have junk
     # characters from the previous message.
-    empty_str = ' ' * 10
-    aws_dependencies_installation = (
-        'pip list | grep boto3 > /dev/null 2>&1 || pip install '
-        'botocore>=1.29.10 boto3>=1.26.1; '
-        # Need to separate the installation of awscli from above because some
-        # other clouds will install boto3 but not awscli.
-        'pip list | grep awscli> /dev/null 2>&1 || pip install "urllib3<2" '
-        'awscli>=1.27.10 "colorama<0.4.5" > /dev/null 2>&1')
-    setup_clouds: List[str] = []
+    empty_str = ' ' * 20
+
+    # All python dependencies will be accumulated and then installed in one
+    # command at the end. This is very fast if the packages are already
+    # installed, so we don't check that.
+    python_packages: Set[str] = set()
+
+    step_prefix = prefix_str.replace('<step>', str(len(commands) + 1))
+    commands.append(f'echo -en "\\r{step_prefix}uv{empty_str}" &&'
+                    'export PATH="$PATH:$HOME/.local/bin" &&'
+                    f'uv -V > /dev/null 2>&1 ||'
+                    'curl -LsSf https://astral.sh/uv/install.sh 2>/dev/null |'
+                    'UV_INSTALL_DIR="$HOME/.local/bin" sh >/dev/null 2>&1')
```

Comment:
Just curious, should we just use uv setup in our provision setup, i.e. use `{constants.UV_PIP_CMD}` directly?

Reply:
Also, we use `{{ sky_activate_python_env }}` in jobs controller yml to make sure pip install goes to the separate env we created. Will the new changes cause any issue?

Reply:
We probably could use SKY_UV_PIP_CMD - I didn't because the existing code wasn't using SKY_PIP_CMD.
Activating the python env normally should work correctly with uv.